src/main/java/edu/kit/hci/soli/filters/DisabledUsersFilter.java

```diff
@@ -0,0 +1,51 @@
+package edu.kit.hci.soli.filters;
+
+import edu.kit.hci.soli.domain.User;
+import edu.kit.hci.soli.service.UserService;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+@Component
+@Order(1)
+@Slf4j
+public class DisabledUsersFilter extends OncePerRequestFilter {
+    private final UserService userService;
+
+    public DisabledUsersFilter(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, @NotNull HttpServletResponse response, @NotNull FilterChain filterChain) throws ServletException, IOException {
+        if (request.getUserPrincipal() == null) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        if (request.getRequestURI().equals("/disabled")) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        User user = userService.findByUserId(request.getUserPrincipal().getName());
```

Comment:
Have you tried just using the Principal directly? Since we provide our own, it should contain the User.

Reply:
Unfournately I get a cast error when I retrieving the as shown below: 
![image](https://github.com/user-attachments/assets/2b6dc2f7-c9c4-4d53-bd0d-1a8473769428)

Reply:
Unfortunately I get a cast error when trying this: 
![image](https://github.com/user-attachments/assets/4827375c-3c17-4c0c-a442-6bae2d3b345b)

Reply:
"Something is not as it should be, therefore you cannot access this" sounds like an error to me. After all, 403 is an error too.

What do you suggest as an alternative?

Reply:
> Unfortunately I get a cast error when trying this

Have you tried running this in a debugger and checking what the type is?
Also: using `instanceof` is probably nicer.
I don't know what error you got, so I can't help you there.

Reply:
> What do you suggest as an alternative?

My suggestion was moving this to the unified `known_error` template